### PR TITLE
Add support for the JSON-B Serialization of Vert.x JSON Object/Array

### LINKED ISSUE #8830: Vert.x JSON wrapped in a Multi is not working as expected
**Describe the bug**
(Describe the problem clearly and concisely.)
When the JSON content is wrapped in a 'Multi', it is not recognised as a Quarkus web resource, instead it embeds the entire json object in a new object called 'map'.

Here is the code:
```
@Path("/users/v1")
@Produces(MediaType.APPLICATION_JSON)
public class UserResource {
@GET
@Path("{name}/multi")
@Produces(MediaType.APPLICATION_JSON)
public Multi<JsonObject> multi(@PathParam String name) {
    Multi<JsonObject> multi = Multi.createFrom().items(new JsonObject().put("Hello", name));
    return multi;
}
@GET
@Path("{name}/uni")
@Produces(MediaType.APPLICATION_JSON)
public Uni<JsonObject> uni(@PathParam String name) {
    Uni<JsonObject> multi = Uni.createFrom().item(new JsonObject().put("Hello", name));
    return multi;
}
}
```
when I access  this end point:
localhost:82/users/v1/all/uni
I 
it returns a nicely formated json like this:
```
{
    "Hello": "all"
}
```

however, when i access  localhost:82/users/v1/all/multi
```
[
    {
        "empty": false,
        "map": {
            "Hello": "all"
        }
    }
]
```

**Expected behavior**
(Describe the expected behavior clearly and concisely.)
It is expected  to return this :
```
{
    "Hello": "all"
}
```

**Actual behavior**
(Describe the actual behavior clearly and concisely.)

It returns this:
```
[
    {
        "empty": false,
        "map": {
            "Hello": "all"
        }
    }
]
```
**To Reproduce**
1. Create a JAX-RS resource method that returns a Multi<JsonObject>
2. Annotate the method with @Produces(MediaType.APPLICATION_JSON) on the methos
3. Add  Multi<JsonObject> multi = Multi.createFrom().items(new JsonObject().put("Hello", name));
    return multi; to the bosy of the method

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by cescoffier**
Fix #8830

### CODE REVIEW

File: extensions/resteasy-jsonb/deployment/pom.xml [hunk 1]
```diff
@@ -26,6 +26,22 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
```
Comment (gsmet) [inline extensions/resteasy-jsonb/deployment/pom.xml:42]: Why do you add that one here and in the other pom?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/resteasy-jsonb/deployment/pom.xml (hunk 1 above)
Comment (cescoffier) [reply extensions/resteasy-jsonb/deployment/pom.xml:42]: There are some methods from Vert.x JSON Object (used in term) that requires Jackson (to rebuild a JSON Object from a String for instance).  It's only used in tests as I've been careful when implementing the serializers/deserializers to NOT use these methods.

Comment (gsmet) [reply extensions/resteasy-jsonb/deployment/pom.xml:42]: What bugs me a bit is that we can't really be sure that we don't need Jackson around for it to work.

Comment (cescoffier) [reply extensions/resteasy-jsonb/deployment/pom.xml:42]: No, we don't , it's the same mechanism that is used for Vert.x. Some method requires Jackson, but we don't require Jackson. If the user uses these methods, then he will have to add the Jackson extension. 

Not the tests in resteasy-mutiny that does NOT depend on Jackson and works.

Comment (gsmet) [reply extensions/resteasy-jsonb/deployment/pom.xml:42]: Can we talk about it on Zulip on Monday? I'm not sure I fully understand what you're saying.

Comment (cescoffier) [reply extensions/resteasy-jsonb/deployment/pom.xml:42]: We're Tuesday, but works for me :-)

**Review by gsmet — APPROVED** [review]

**Pull request merged**